applications/TREX_Network.md

```diff
@@ -0,0 +1,284 @@
+
+- TREX API library ([GitHub repo](https://github.com/NexTokenTech/trex-subxt))
+
+Tested and successfully submitted dTRE data to the TREX blockchain node.
+
+## Development Roadmap :nut_and_bolt:
+
+Since the basic protocol and consensus have been implemented in the current [TREX node repo](https://github.com/NexTokenTech/TREX). Our milestones focus on delivering support to integrate with dApps that use the dTRE technology and land the first dTRE service to the market.
+
+
+### Milestone 1 — XCMP Integration
+
+- **Estimated Duration:** 1 month
+- **FTE:**  2
+- **Costs:** 10,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 0a. | License | MIT License
+| 0b. | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a parachain can route a dTRE message to the TREX network.|
+| 0c. | Testing Guide | Core functions will be fully covered by unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| 0d. | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 0e. | Article | We will publish an online blog that explains how this module works as described in the following figure.
+| 1. | Substrate module: XCMP | Provide run-time pallet to process cross-chain requests to put dTRE data on the TREX blockchain.
+| 2. | TREX node update | The TREX node is updated to provide any necessary low-level supports for XCMP protocal and cross-chain dTRE data.
+
+### Milestone 2 — Implement TREX Mining Pool Server and Client Software
+
+- **Estimated duration:** 2 months
+- **FTE:**  2
+- **Costs:** 20,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 0a. | License | MIT License
+| 0b. | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a content creator can generate tradable content and a user can gain early access to this content with a fee. |
+| 0c. | Testing Guide | Core functions will be fully covered by unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| 0d. | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 0e. | Article | We will publish an online blog that explains how this module works as described in the following figure.
+| 1. | TREX mining server | A TREX node which host a mining pool and access connection from remote miners and send rewards to miners. To protect the consensus against attacks from cluter computers, the mining pool needs to host a large hash table so that the mining efficiency is close to the cluster architecture coordinated by a center.
```

Comment:
```suggestion
| 1. | TREX mining server | A TREX node which host a mining pool and access connection from remote miners and send rewards to miners. To protect the consensus against attacks from computer clusters, the mining pool needs to host a large hash table so that the mining efficiency is close to the cluster architecture coordinated by a center.
```

Reply:
Hi, I think what I descrbied is not about network consensus but encryption strength. The timed-release encryption is basically a crypto system which allow user to encrpyt data and decrypt it automatically in predetermined future. What I mean is you need to get more computing power than the network to get earlier access independently to the encrypted contents than everyone else in the network. This is not about to take over the blockchain but decrypt the encrypted contents. For the consensus, you need to have 51% or more hash power to take the blockchain, which is the same as other PoW blockchains.

For the ASIC-resistancy, it is about its hash function. Our algorithm can interface with different hash function, if some hash function have ASIC to accelerate, we could change to other hash function which is ASIC-resistant. The hash function is essentially providing peusdo-randomness in building the graph and finding the collision to solve the puzzle (which is the encryption scheme), thus, many hash function works with our algorithm framework (e. g. SHA256, BLAKE3, etc). I am working on some documentation to explain more about the algorithm. I could also explain it to you if you are interested.